Odata-docs/performance/disable-writer-validations.md

```diff
@@ -0,0 +1,130 @@
+---
+title:  "Disable writer validations"
+description: A guide on how to improve performance by disabling writer validations.
+date:   2024-01-23
+ms.date: 1/23/2023
+author: habbes
+ms.author: clhabins
+---
+
+# Disable writer validations
+
+**Applies To**:[!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v8.md)][!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v7.md)][!INCLUDE[appliesto-odatalib](../includes/appliesto-odatalib-v7.md)]
+
+During serialization, the OData writer performs various validations to ensure that the payload is valid (e.g. no duplicate properties) and conforms to the schema (e.g. property type matches the entity type's definition). These validations introduce some overhead to the serialization process. If you know ahead of time that the payload you're writing is correct, you can disable one or more validations to improve performance.
+
+Use the [`ODataMessageWriterSettings.Validations`](/dotnet/api/microsoft.odata.odatamessagewritersettings.validations) property to configure which validations you want to enable or disable.
+
+You can disable all validations using [`ValidationKinds.None`]:
+
+```csharp
+messageWriterSettings.Validations = ValidationKinds.None;
+```
+
+You can selectively enable validations using the `&` operator:
+
+```csharp
+messageWriterSettings.Validations = ValidationKinds.ThrowOnDuplicatePropertyNames & ValidationKinds.ThrowIfTypeConflictsWithMetadata
+```
+
+You can enable all validations using [`ValidationKinds.All`]:
+
+```csharp
+messageWriterSettings.Validations = ValidationKinds.All;
+```
+
+To learn more about different validation options, consult the [`ValidationKinds` enum documentation](/dotnet/api/microsoft.odata.validationkinds).
+
+## Disabling writer validations when using `Microsoft.OData.Core` directly
+
+When using the core library directly, you pass the `ODataMessageWriterSettings` directly to the [`ODataMessageWriter`](/dotnet/api/microsoft.odata.odatamessagewriter) construtor.
+
+```csharp
+ODataMessageWriterSettings settings = new ODataMessageWriterSettings
+{
+    ODataUri = new ODataUri() { /* ... */ },
+    Validations = ValidationKinds.None
+};
+
+ODataMessageWriter writer = new ODataMessageWriter(odataMessage, settings, edmModel);
+
+// ...
+```
+
+## Disabling writer validations when using `Microsoft.AspNetCore.OData` 8
+
+In `Microsoft.AspNetCore.OData` 8, you can pass a service configuration action as the third argument of `AddRouteComponents`. Use
+this to inject a custom request-scoped `ODataMessageWriterSettings`, with validations disabled, to the service container:
+
+```csharp
+services
+.AddControllers()
+.AddOData(options =>
+    options.AddRouteComponents("routePrefix", edmModel, container =>
```

Comment:
Using named parameters would make this snippet self-describing and easier to understand, i.e.,
```csharp
routePrefix: "...",
model: edmModel,
configureAction: container => // ...
```